internal/analytics/tilt_analytics_test.go

```diff
@@ -0,0 +1,223 @@
+	os.opt = opt
+	return nil
+}
+
+func TestCount(t *testing.T) {
+	for _, test := range testCases(true, false, false) {
+		t.Run(test.name, func(t *testing.T) {
+			ma := analytics.NewMemoryAnalytics()
+			os := &optSetting{}
+			a := NewTiltAnalytics(test.opt, os.setOpt, ma)
+			a.Count("foo", map[string]string{}, 1)
+			var expectedCounts []analytics.CountEvent
+			if test.expectRecord {
+				expectedCounts = append(expectedCounts, analytics.CountEvent{
+					Name: "foo",
+					Tags: map[string]string{},
+					N:    1,
+				})
+			}
+			assert.Equal(t, expectedCounts, ma.Counts)
+		})
+	}
+}
+
+func TestIncr(t *testing.T) {
+	for _, test := range testCases(true, false, false) {
+		t.Run(test.name, func(t *testing.T) {
+			ma := analytics.NewMemoryAnalytics()
+			os := &optSetting{}
+			a := NewTiltAnalytics(test.opt, os.setOpt, ma)
+			a.Incr("foo", map[string]string{})
+			var expectedCounts []analytics.CountEvent
+			if test.expectRecord {
+				expectedCounts = append(expectedCounts, analytics.CountEvent{
+					Name: "foo",
+					Tags: map[string]string{},
+					N:    1,
+				})
+			}
+			assert.Equal(t, expectedCounts, ma.Counts)
+		})
+	}
+}
+
+func TestTimer(t *testing.T) {
+	for _, test := range testCases(true, false, false) {
+		t.Run(test.name, func(t *testing.T) {
+			ma := analytics.NewMemoryAnalytics()
+			os := &optSetting{}
+			a := NewTiltAnalytics(test.opt, os.setOpt, ma)
+			a.Timer("foo", time.Second, map[string]string{})
+			var expectedTimes []analytics.TimeEvent
+			if test.expectRecord {
+				expectedTimes = append(expectedTimes, analytics.TimeEvent{
+					Name: "foo",
+					Tags: map[string]string{},
+					Dur:  time.Second,
+				})
+			}
+			assert.Equal(t, expectedTimes, ma.Timers)
+		})
+	}
+}
+
+func TestIncrIfUnopted(t *testing.T) {
+	for _, test := range testCases(false, false, true) {
+		t.Run(test.name, func(t *testing.T) {
+			ma := analytics.NewMemoryAnalytics()
+			os := &optSetting{}
+			a := NewTiltAnalytics(test.opt, os.setOpt, ma)
+			a.IncrIfUnopted("foo", map[string]string{})
+			var expectedCounts []analytics.CountEvent
+			if test.expectRecord {
+				expectedCounts = append(expectedCounts, analytics.CountEvent{
+					Name: "foo",
+					Tags: map[string]string{},
+					N:    1,
+				})
+			}
+			assert.Equal(t, expectedCounts, ma.Counts)
+		})
+	}
+}
+
+func analyticsViaTransition(t *testing.T, initialOpt, newOpt analytics.Opt) (*TiltAnalytics, *analytics.MemoryAnalytics) {
+	ma := analytics.NewMemoryAnalytics()
+	os := optSetting{}
+	a := NewTiltAnalytics(initialOpt, os.setOpt, ma)
+	switch newOpt {
+	case analytics.OptIn:
+		err := a.OptIn()
+		if err != nil {
+			assert.FailNow(t, err.Error())
+		}
+	case analytics.OptOut:
+		err := a.OptOut()
+		if err != nil {
+			assert.FailNow(t, err.Error())
+		}
+	}
+
+	if !assert.Equal(t, newOpt, os.opt) {
+		t.FailNow()
+	}
+
+	// wipe out the reports of opt-in/out, since those are just setup for another test
```

Comment:
...huh?

Reply:
ah, i think you probs want: 
```suggestion
	// wipe out the reports of opt-in/out, since we're testing those separately
```